src/application.rs

```diff
@@ -516,6 +613,48 @@ impl Application {
                     self.ui.show_ip_dialog(iface_data);
                 }
             }
+            UiActions::ChangeServer => {
+                let is_onboarded = self.model.borrow().node_status.is_onboarded();
+
+                if is_onboarded {
+                    self.ui.message_box(
+                        "WARNING",
+                        "The node is onboarded and the server URL cannot be changed.",
+                    );
+                } else {
+                    let url = self
+                        .model
+                        .borrow()
+                        .node_status
+                        .server
+                        .clone()
+                        .unwrap_or_default();
+                    self.ui.show_server_url_dialog(&url);
+                }
+            }
+            UiActions::AppAction(app_action) => match app_action {
+                MonActions::NetworkInterfaceUpdated(old, new) => {
+                    debug!("Setting DPC for {}", &old.iface_name);
+                    debug!("OLD DPC: {:#?}", &old);
+                    debug!("NEW DPC: {:#?}", &new);
+                    if old == new {
+                        debug!("Not changed, not sending DPC");
+                    } else {
+                        self.send_dpc(old, new);
+                    }
+                    self.ui.pop_layer();
+                }
+                MonActions::ServerUpdated(url) => {
+                    debug!("Setting server URL to: {}", &url);
+                    self.send_ipc_message(
+                        IpcMessage::new_request(Request::SetServer(url.clone())),
+                        move |app| {
+                            app.model.borrow_mut().node_status.server = Some(url.clone());
```

Comment:
clone is not needed, it's the last place where you use url variable in this branch